commands/apps_test.go

```diff
@@ -45,3 +47,137 @@ func TestCommandsAppsCreateAttemptsToValidateStackOnError(t *testing.T) {
 	assert.Error(err)
 	assert.Regexp("bad request: unable to find stack", err)
 }
+
+// Looks like nested functions are not yet supported by the compiler
+func OverwriteFile(loc string, data string) (err error) {
+	err = os.Remove(loc)
+	if err != nil {
+		log.Println("[ERROR] unable to remove files, perhaps they do not exist?")
+	}
+	f, err := os.Create(loc)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(data)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func TestInitNodejsBasicAppEncounteringPossibleFailureStates(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup
+	var testCases = []struct {
+		servConf string
+		pkgJson  string
+		isFatal  bool
+	}{
+		{`location / {
+			proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+			proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
+			proxy_set_header Host $host;
+			include /etc/nginx/section.module/node.conf;
+		}
+		
+		location ~ "/next-proxy-hop/" {
+			proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+			proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
+			proxy_set_header Host $host;
+			proxy_pass http://next-hop;
+		}`, `{
+			"name": "api-routes",
+			"version": "1.0.0",
+			"scripts": {
+				"dev": "next",
+				"build": "next build",
+				"start": "next start -p 8080",
+				"predeploy": "npm install && npm run build",
+				"deploy": "sectionctl deploy -a 1887 -i 7749"
+			},
+			"dependencies": {
+				"next": "latest",
+				"react": "^16.8.6",
+				"react-dom": "^16.8.6",
+				"swr": "0.1.18"
+			},
+			"license": "MIT"
+		}`, false}, // no files are broken or missing
+		{`proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
+		proxy_set_header Host $host;
+		include /etc/nginx/section.module/node.conf;
+	}
+	
+	location ~ "/next-proxy-hop/" {
+		proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+		proxy_set_header X-Forwa`, `{
+			"name": "api-routes",
+			"version": "1.0.0",
+			"scripts": {
+				"dev": "next",
+				"build": "next build",
+				"predeploy": "npm install && npm run build",
+				"deploy": "sectionctl deploy -a 1887 -i 7749"
+			},
+			"dependencies": {
+				"next": "latest",
+				"react": "^16.8.6",
+				"react-dom": "^16.8.6",
+				"swr": "0.1.18"
+			},
+			"license": "MIT"
+		}`, false}, // both files are broken
+		{``, ``, false}, // both files are empty
+		{``, `{
+			"name": "api-routes",
+			"version": "1.0.0",
+			"scripts": {
+				"dev": "next",
+				"build": "next build"
+				"predeploy": "npm install887 -i 7749"
+			"dependencies": {
+				"next": "latest",
+				"react": "^16.8.6",
+				"react-dom": "^16.8.6",
+				"swr": "0.1.18"
+			},
+			"license": "MIT"`, true}, // package.json can not be unmarshalled
+	}
+
+	cmd := AppsInitCmd{
+		StackName: "nodejs-basic",
+		Force:     false,
+	}
+	for _, tc := range testCases {
+		n := ""
+		t.Run(n, func(t *testing.T) {
+			// Setup
+			err := OverwriteFile("server.conf", tc.servConf)
```

Comment:
There are a few big drawbacks to this approach:

- If a test in the table fails, the data fed into the test will be blown away by subsequent tests in the table that are run (because we're reading and writing the same file in each test case)
- If I'm writing a test and I decide to overwrite `sectionctl.go` or `api/api.go`, I'm gonna have a bad time. 

The better way to handle this is to create a temporary directory for the test case to run in. 

There's a [good example of this](https://github.com/section/sectionctl/blob/583716bfef5195e3d7ae3b54031cf2a7be81c89f/commands/deploy_test.go#L109-L111) in the `deploy` subcommand tests.